numpy_financial/_financial.py

```diff
@@ -825,6 +827,40 @@ def irr(values, *, guess=None, tol=1e-12, maxiter=100, raise_exceptions=False):
     return np.nan
 
 
+@nb.njit(parallel=True)
+def _npv(rates, cashflows, result):
+    r"""Native version of the ``npv`` function.
+
+    Warnings
+    --------
+    For internal use only, note that this function performs no error checking.
+    """
+
+    # TODO: Is using `prange` actually faster here?
+    for i in prange(rates.shape[0]):
+        for j in prange(cashflows.shape[0]):
+            acc = 0.0
+            for t in range(cashflows.shape[1]):
+                acc += cashflows[j, t] / ((1.0 + rates[i]) ** t)
+            result[i, j] = acc
+
+
+@nb.jit(forceobj=True)  # Need ``forceobj`` to support decimal.Decimal
+def _npv_decimal(rates, cashflows, result):
+    r"""Version of the ``npv`` function supporting ``decimal.Decimal`` types
+
+    Warnings
+    --------
+    For internal use only, note that this function performs no error checking.
+    """
+    for i in range(rates.shape[0]):
+        for j in range(cashflows.shape[0]):
+            acc = Decimal("0.0")
+            for t in range(cashflows.shape[1]):
+                acc += cashflows[j, t] / ((Decimal("1.0") + rates[i]) ** t)
+            result[i, j] = acc
```

Comment:
This is essentially a copy of `_npv` with the constants replaced with `decimal.Decimal`s. I've tried doing something along the lines of `dtype = Decimal if arr.dtype = np.dtype("O") else arr.dtype` but this wouldn't compile in `nopython` mode. I'm trying to think of a neater way to implement this but haven't been able to think of one yet